arthur/tasks.py

```diff
@@ -241,6 +241,32 @@ def get(self, task_id):
 
         return task
 
+    def update(self, task_id, task):
+        """Update a task in the registry.
+
+        Update a task store in the registry using its task identifier. When
+        the task does not exist, a `NotFoundError` exception will be
+        raised.
+
+        :param task_id: task identifier
+        :param task: task object
+
+        :returns: a task object
+
+        :raises NotFoundError: raised when the requested task is not
+            found on the registry
+        """
+        self._rwlock.reader_acquire()
+
+        if task_id not in [k.decode("utf-8") for k in self.conn.keys()]:
+            self._rwlock.reader_release()
+            raise NotFoundError(element=str(task_id))
+
+        self.conn.set(task_id, pickle.dumps(task))
```

Comment:
If this raises an exception, other processes might get stuck because the lock is never released.